mlrun/frameworks/langchain/model_server.py

```diff
@@ -0,0 +1,152 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# flake8: noqa  - this is until we take care of the F401 violations with respect to __all__ & sphinx
+
+from typing import Any, Dict, Union
+
+import langchain_community.llms
+from langchain_core.language_models.llms import LLM
+
+import mlrun
+from mlrun.serving.v2_serving import V2ModelServer
+
+
+class LangChainModelServer(V2ModelServer):
+    """
+    LangChain Model serving class, inheriting the V2ModelServer class for being initialized automatically by the model
+    server and be able to run locally as part of a nuclio serverless function, or as part of a real-time pipeline.
+    For compatibility with langchain models, this class supports the following operations:
+    - invoke
+    - batch
+    - ainvoke
+    - abatch
+    in the future we will add streaming support.
+    Notice: In order to use this serving class, please ensure that the langchain_community package is installed.
+    """
+
+    def __init__(
+        self,
+        context: mlrun.MLClientCtx = None,
+        model_class: str = None,
+        llm: Union[str, LLM] = None,
+        init_kwargs: Dict[str, Any] = None,
+        generation_kwargs: Dict[str, Any] = None,
+        name: str = None,
+        model_path: str = None,
+        **kwargs,
+    ):
+        """
+        Initialize a serving class for general llm usage.
+        :param context:           The mlrun context to use.
+        :param model_class:       The class of the model to use.
+        :param llm:               The llm object itself in case of local usage.
+        :param init_kwargs:       The initialization arguments to use while initializing the llm.
+        :param generation_kwargs: The generation arguments to use while generating text.
+        :param name:              The name of this server to be initialized.
+        :param model_path:        Not in use. When adding a model pass any string value
+        """
+        super().__init__(name=name, context=context, model_path=model_path)
+        self.model_class = model_class
+        self.llm = llm
+        self.init_kwargs = init_kwargs or {}
+        self.generation_kwargs = generation_kwargs
+
+    def load(self):
+        """
+        loads the model.
+        """
+        # If the llm is already an LLM object, use it directly
+        if isinstance(self.llm, LLM):
+            self.model = self.llm
+            return
+        # If the llm is a string (or not given, then we take default model), load the llm from langchain.
```

Comment:
how you take a default model?

Reply:
you just init without specifing (ex: openai() and then you get gpt-3.5)